### DISCUSSION TITLE
How to assign border-color using props?

### DISCUSSION THREAD

**Discussion by hrshpreet** [tailwindlabs/tailwindcss#12648] [Help] [answered]
Tailwind colors do not work when I pass them as props but when I write them directly. Below are some files and screenshots for reference:

tailwind.config.js:
<img width="374" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/93072998/e1b9ea52-57a0-4099-a4d6-2ef33d15b65f">

Code:
<img width="804" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/93072998/068ff825-12e0-4565-b0f2-f417740c06ce">

Result:
<img width="949" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/93072998/3b7f66ce-e805-45bd-9045-a11418257646">
I expect it to have a border color

Previously i had tried the same without props but by making an array but that didn't work as well: https://github.com/hrshpreet/portfolio/blob/main/src/components/techstack.jsx

Ps. I have tried console.log'ing the props and they work properly, but don't show up in the result.

**Comment by wongjn** [comment, marked as answer]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

**Reply by hrshpreet** [reply]
Cant thank you enough; I was struggling with that for a couple of days. Thanks a lot budyy.
